### PR TITLE
[JENKINS-71015] Re-Support Job DSL

### PR DESCRIPTION
**Pull request by limejuny**
### Detail

 - After active-choices-plugin 2.6.0, it use `ScriptlerBuilder` instead of `scriptlerScriptId` and `parameter`. But `ScriptlerBuilder` doesn't have `DataBoundConstructor` annotation, so Job DSL can't recognize the scriptler script syntax in `cascadeChoiceParameter`.
 - Before 2.5.7
   - image
   - ```
     scriptlerScript {
       scriptlerScriptId(String value)
       parameters {
         scriptlerScriptParameter {
           name(String value)
         }
       }
     }
     ```
 - After 2.6.0
   - ![image](https://github.com/biouno/uno-choice-plugin/assets/16630665/cec02a53-486f-424b-bebe-c25977c1a46a)
 - After this commit
   - ![image](https://github.com/jenkinsci/scriptler-plugin/assets/16630665/8036bfa7-777d-4a2d-9ae8-d75b470bae47)
   - ```
     scriptlerScript {
       scriptlerBuilder {
         buildersId(String value)
         scriptId(String value)
         propagateParams(boolean value)
         parameters {
           parameter {
             name(String value)
             value(String value)
           }
         }
       }
       isSandboxed(Boolean value)
       scriptlerScriptId(String value)
     }
     ```

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Review by mtughan — CHANGES_REQUESTED** [review]

File: src/main/java/org/jenkinsci/plugins/scriptler/builder/ScriptlerBuilder.java [hunk 1]
```diff
@@ -88,6 +90,11 @@ public ScriptlerBuilder(String builderId, String scriptId, boolean propagatePara
         this.propagateParams = propagateParams;
     }
 
+    @DataBoundConstructor
```
[the remark this answers is not in the document]
Comment (limejuny) [reply src/main/java/org/jenkinsci/plugins/scriptler/builder/ScriptlerBuilder.java:93]: Yes, I think it is more complicated, too...
But I found the `definedParams` in other json file or test code.

I didn't deep dive other part of codes, so I don't know why is the `definedParams` exists, and I want to change as small as possible.
So I created new constructor of `ScriptlerBuilder`.

Here is some examples.

---
image
---
image

[the remark this answers is not in the document]
Comment (mtughan) [reply src/main/java/org/jenkinsci/plugins/scriptler/builder/ScriptlerBuilder.java:93]: No worries. I would take it out, as this `defineParams` key is only used in the REST communication between the Jelly in the UI and the backend. Long-term, I would probably like to get rid of it, and there's no reason to introduce its usage into a user-facing part of the code (which the Job DSL is).

[the remark this answers is not in the document]
Comment (limejuny) [reply src/main/java/org/jenkinsci/plugins/scriptler/builder/ScriptlerBuilder.java:93]: OK, Thanks. I'll test again without `defineParams`.

[the remark this answers is not in the document]
Comment (limejuny) [reply src/main/java/org/jenkinsci/plugins/scriptler/builder/ScriptlerBuilder.java]: It works with this script, thanks!

---
image
---
image

**Review by mtughan — CHANGES_REQUESTED** [review]
The code diff looks good now, but the commit message still shows the outdated `defineParams` syntax. Can you clean that up as well? Would also be helpful to fix the PR description to match.

File: src/main/java/org/jenkinsci/plugins/scriptler/builder/ScriptlerBuilder.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mtughan) [reply src/main/java/org/jenkinsci/plugins/scriptler/builder/ScriptlerBuilder.java]: Label added. :)

[the remark this answers is not in the document]
Comment (limejuny) [reply src/main/java/org/jenkinsci/plugins/scriptler/builder/ScriptlerBuilder.java]: Thanks!

**Comment by limejuny** [comment]
> The code diff looks good now, but the commit message still shows the outdated `defineParams` syntax. Can you clean that up as well? Would also be helpful to fix the PR description to match.

Oh, Thank you for the good point.

**Review by mtughan — APPROVED** [review]

**Comment by mtughan** [comment]
Oops, sorry for the delay on this one, @limejuny! I was waiting for the checks to pass and it slipped my notice afterwards.

**Pull request merged**